```
10:47:19 Using database oc_autotest3
10:47:19 Setup environment for mysqlmb4 testing on local storage ...
10:47:19 Fire up the mysql docker
10:47:20 Waiting for MySQL(utf8mb4) initialisation ...
10:57:22 ........................................................................................................................................................................................................
10:57:22 [ERROR] Waited 600 seconds, no response
10:57:22 Kill the docker 9b7dee09bda11058a69097640488fc4b31fde0e70c0f8a395416367bdce21356
10:57:22 9b7dee09bda11058a69097640488fc4b31fde0e70c0f8a395416367bdce21356
10:57:22 9b7dee09bda11058a69097640488fc4b31fde0e70c0f8a395416367bdce21356
10:57:22 Makefile:177: recipe for target 'test-php' failed
```

--- a/tests/integration/config/behat.yml
+++ b/tests/integration/config/behat.yml
@@ -20,8 +20,6 @@ default:
             baseUrl: http://localhost:8080
         - CalDavContext:
             baseUrl: http://localhost:8080
-        - ChecksumsContext:
-            baseUrl: http://localhost:8080
         - CommandLineContext:
             baseUrl: http://localhost:8080
             ocPath: ../../

--- a/tests/integration/features/bootstrap/BasicStructure.php
+++ b/tests/integration/features/bootstrap/BasicStructure.php
@@ -9,6 +9,7 @@ trait BasicStructure {
 
 	use Auth;
 	use Trashbin;
+	use Checksums;
 
 	/** @var string */
 	private $currentUser = '';

--- a/tests/integration/features/bootstrap/Checksums.php
+++ b/tests/integration/features/bootstrap/Checksums.php
@@ -29,26 +29,13 @@ trait Checksums {
 	 * @param string $checksum
 	 */
 	public function userUploadsFileToWithChecksum($user, $source, $destination, $checksum) {
-		$client = new Client();
 		$file = \GuzzleHttp\Stream\Stream::factory(fopen($source, 'r'));
-		try {
-			$this->response = $client->put(
-				substr($this->baseUrl, 0, -4) . $this->davPath . $destination,
-				[
-					'auth' => [
-						$user,
-						$this->getPasswordForUser($user)
-					],
-					'body' => $file,
-					'headers' => [
-						'OC-Checksum' => $checksum
-					]
-				]
-			);
-		} catch (\GuzzleHttp\Exception\ServerException $e) {
-			// 4xx and 5xx responses cause an exception
-			$this->response = $e->getResponse();
-		}
+		$this->response = $this->makeDavRequest($user,
+							  'PUT',
+							  $destination,
+							  ['OC-Checksum' => $checksum],
+							  $file,
+							  "files");
 	}
 
 	/**
@@ -71,7 +58,7 @@ trait Checksums {
 		$client = new Client();
 		$request = $client->createRequest(
 			'PROPFIND',
-			substr($this->baseUrl, 0, -4) . $this->davPath . $path,
+			substr($this->baseUrl, 0, -4) . $this->getDavFilesPath($user) . $path,
 			[
 				'body' => '<?xml version="1.0"?>
 <d:propfind  xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
@@ -107,24 +94,6 @@ trait Checksums {
 		if ($checksums['value'][0]['value'] !== $checksum) {
 			throw new \Exception("Expected $checksum, got ".$checksums['value'][0]['value']);
 		}
-	}
-
-	/**
-	 * @When user :user downloads the file :path
-	 * @param string $user
-	 * @param string $path
-	 */
-	public function userDownloadsTheFile($user, $path) {
-		$client = new Client();
-		$this->response = $client->get(
-			substr($this->baseUrl, 0, -4) . $this->davPath . $path,
-			[
-				'auth' => [
-					$user,
-					$this->getPasswordForUser($user),
-				]
-			]
-		);
 	}
 
 	/**
@@ -201,24 +170,16 @@ trait Checksums {
 	 * @param string $destination
 	 * @param string $checksum
 	 */
-	public function userUploadsChunkFileOfWithToWithChecksum($user, $num, $total, $data, $destination, $checksum)
-	{
-		$client = new Client();
+	public function userUploadsChunkFileOfWithToWithChecksum($user, $num, $total, $data, $destination, $checksum) {
+		//$client = new Client();
 		$num -= 1;
-		$this->response = $client->put(
-			substr($this->baseUrl, 0, -4) . $this->davPath . $destination . '-chunking-42-'.$total.'-'.$num,
-			[
-				'auth' => [
-					$user,
-					$this->getPasswordForUser($user)
-				],
-				'body' => $data,
-				'headers' => [
-					'OC-Checksum' => $checksum,
-					'OC-Chunked' => '1',
-				]
-			]
-		);
-
+		$data = \GuzzleHttp\Stream\Stream::factory($data);
+		$file = $destination . '-chunking-42-' . $total . '-' . $num;
+		$this->response = $this->makeDavRequest($user,
+							  'PUT',
+							  $file,
+							  ['OC-Checksum' => $checksum, 'OC-Chunked' => '1'],
+							  $data,
+							  "files");
 	}
 }

--- a/tests/integration/features/bootstrap/Checksums.php
+++ b/tests/integration/features/bootstrap/Checksums.php
@@ -5,46 +5,20 @@ require __DIR__ . '/../../../../lib/composer/autoload.php';
 use GuzzleHttp\Client;
 use GuzzleHttp\Message\ResponseInterface;
 
-class ChecksumsContext implements \Behat\Behat\Context\Context {
-	/** @var string  */
-	private $baseUrl;
-	/** @var Client */
-	private $client;
-	/** @var ResponseInterface */
-	private $response;
+trait Checksums {
 
-	/**
-	 * @param string $baseUrl
-	 */
-	public function __construct($baseUrl) {
-		$this->baseUrl = $baseUrl;
-
-		// in case of ci deployment we take the server url from the environment
-		$testServerUrl = getenv('TEST_SERVER_URL');
-		if ($testServerUrl !== false) {
-			$this->baseUrl = substr($testServerUrl, 0, -5);
-		}
-	}
-
-	/** @BeforeScenario */
-	public function setUpScenario() {
-		$this->client = new Client();
-	}
-
-	/** @AfterScenario */
-	public function tearDownScenario() {
-	}
-
+	use Webdav;
 
 	/**
 	 * @param string $userName
 	 * @return string
 	 */
 	private function getPasswordForUser($userName) {
-		if($userName === 'admin') {
-			return 'admin';
+		if ($userName === 'admin') {
+			return $this->adminUser;
+		} else {
+			return $this->regularUser;
 		}
-		return '123456';
 	}
 
 	/**
@@ -54,12 +28,12 @@ class ChecksumsContext implements \Behat\Behat\Context\Context {
 	 * @param string $destination
 	 * @param string $checksum
 	 */
-	public function userUploadsFileToWithChecksum($user, $source, $destination, $checksum)
-	{
+	public function userUploadsFileToWithChecksum($user, $source, $destination, $checksum) {
+		$client = new Client();
 		$file = \GuzzleHttp\Stream\Stream::factory(fopen($source, 'r'));
 		try {
-			$this->response = $this->client->put(
-				$this->baseUrl . '/remote.php/webdav' . $destination,
+			$this->response = $client->put(
+				substr($this->baseUrl, 0, -4) . $this->davPath . $destination,
 				[
 					'auth' => [
 						$user,
@@ -93,11 +67,11 @@ class ChecksumsContext implements \Behat\Behat\Context\Context {
 	 * @param string $user
 	 * @param string $path
 	 */
-	public function userRequestTheChecksumOfViaPropfind($user, $path)
-	{
-		$request = $this->client->createRequest(
+	public function userRequestTheChecksumOfViaPropfind($user, $path) {
+		$client = new Client();
+		$request = $client->createRequest(
 			'PROPFIND',
-			$this->baseUrl . '/remote.php/webdav' . $path,
+			substr($this->baseUrl, 0, -4) . $this->davPath . $path,
 			[
 				'body' => '<?xml version="1.0"?>
 <d:propfind  xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
@@ -111,7 +85,7 @@ class ChecksumsContext implements \Behat\Behat\Context\Context {
 				]
 			]
 		);
-		$this->response = $this->client->send($request);
+		$this->response = $client->send($request);
 	}
 
 	/**
@@ -140,10 +114,10 @@ class ChecksumsContext implements \Behat\Behat\Context\Context {
 	 * @param string $user
 	 * @param string $path
 	 */
-	public function userDownloadsTheFile($user, $path)
-	{
-		$this->response = $this->client->get(
-			$this->baseUrl . '/remote.php/webdav' . $path,
+	public function userDownloadsTheFile($user, $path) {
+		$client = new Client();
+		$this->response = $client->get(
+			substr($this->baseUrl, 0, -4) . $this->davPath . $path,
 			[
 				'auth' => [
 					$user,
@@ -171,22 +145,22 @@ class ChecksumsContext implements \Behat\Behat\Context\Context {
 	 * @param string $source
 	 * @param string $destination
 	 */
-	public function userCopiedFileTo($user, $source, $destination)
-	{
-		$request = $this->client->createRequest(
+	public function userCopiedFileTo($user, $source, $destination) {
+		$client = new Client();
+		$request = $client->createRequest(
 			'MOVE',
-			$this->baseUrl . '/remote.php/webdav' . $source,
+			substr($this->baseUrl, 0, -4) . $this->davPath . $source,
 			[
 				'auth' => [
 					$user,
 					$this->getPasswordForUser($user),
 				],
 				'headers' => [
-					'Destination' => $this->baseUrl . '/remote.php/webdav' . $destination,
+					'Destination' => substr($this->baseUrl, 0, -4) . $this->davPath . $destination,
 				],
 			]
 		);
-		$this->response = $this->client->send($request);
+		$this->response = $client->send($request);
 	}
 
 	/**
@@ -229,9 +203,10 @@ class ChecksumsContext implements \Behat\Behat\Context\Context {
 	 */
 	public function userUploadsChunkFileOfWithToWithChecksum($user, $num, $total, $data, $destination, $checksum)
 	{
+		$client = new Client();
 		$num -= 1;
-		$this->response = $this->client->put(
-			$this->baseUrl . '/remote.php/webdav' . $destination . '-chunking-42-'.$total.'-'.$num,
+		$this->response = $client->put(
+			substr($this->baseUrl, 0, -4) . $this->davPath . $destination . '-chunking-42-'.$total.'-'.$num,
 			[
 				'auth' => [
 					$user,

--- a/tests/integration/features/bootstrap/WebDav.php
+++ b/tests/integration/features/bootstrap/WebDav.php
@@ -64,6 +64,7 @@ trait WebDav {
 		} else {
 			$options['auth'] = [$user, $this->regularUser];
 		}
+
 		$request = $client->createRequest($method, $fullUrl, $options);
 		if (!is_null($headers)){
 			foreach ($headers as $key => $value) {
@@ -197,6 +198,19 @@ trait WebDav {
 	public function downloadingFile($fileName) {
 		try {
 			$this->response = $this->makeDavRequest($this->currentUser, 'GET', $fileName, []);
+		} catch (\GuzzleHttp\Exception\ClientException $e) {
+			$this->response = $e->getResponse();
+		}
+	}
+
+	/**
+	 * @When user :user downloads the file :fileName
+	 * @param string $user
+	 * @param string $fileName
+	 */
+	public function userDownloadsTheFile($user, $fileName) {
+		try {
+			$this->response = $this->makeDavRequest($user, 'GET', $fileName, []);
 		} catch (\GuzzleHttp\Exception\ClientException $e) {
 			$this->response = $e->getResponse();
 		}

--- a/tests/integration/features/bootstrap/WebDav.php
+++ b/tests/integration/features/bootstrap/WebDav.php
@@ -563,7 +563,7 @@ trait WebDav {
 	 * @param string $data
 	 * @param string $destination
 	 */
-	public function userUploadsChunkFileOfWithToWithChecksum($user, $num, $total, $data, $destination)
+	public function userUploadsChunkedFile($user, $num, $total, $data, $destination)
 	{
 		$num -= 1;
 		$data = \GuzzleHttp\Stream\Stream::factory($data);

--- a/tests/integration/features/checksums.feature
+++ b/tests/integration/features/checksums.feature
@@ -24,7 +24,7 @@ Feature: checksums
     Given using old dav path
     And user "user0" exists
     And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
-    When User "user0" moved file "/myChecksumFile.txt" to "/myMovedChecksumFile.txt"
+    When User "user0" moves file "/myChecksumFile.txt" to "/myMovedChecksumFile.txt"
     And user "user0" request the checksum of "/myMovedChecksumFile.txt" via propfind
     Then The webdav checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
 
@@ -32,24 +32,8 @@ Feature: checksums
     Given using old dav path
     And user "user0" exists
     And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
-    When User "user0" moved file "/myChecksumFile.txt" to "/myMovedChecksumFile.txt"
+    When User "user0" moves file "/myChecksumFile.txt" to "/myMovedChecksumFile.txt"
     And user "user0" downloads the file "/myMovedChecksumFile.txt"
-    Then The header checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
-
-  Scenario: Copying a file with checksum should return the checksum in the propfind
-    Given using old dav path
-    And user "user0" exists
-    And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
-    When User "user0" copied file "/myChecksumFile.txt" to "/myChecksumFileCopy.txt"
-    And user "user0" request the checksum of "/myChecksumFileCopy.txt" via propfind
-    Then The webdav checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
-
-  Scenario: Copying file with checksum should return the checksum in the download header
-    Given using old dav path
-    And user "user0" exists
-    And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
-    When User "user0" copied file "/myChecksumFile.txt" to "/myChecksumFileCopy.txt"
-    And user "user0" downloads the file "/myChecksumFileCopy.txt"
     Then The header checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
 
   Scenario: Uploading a chunked file with checksum should return the checksum in the propfind
@@ -72,6 +56,84 @@ Feature: checksums
 
   Scenario: Downloading a file from local storage has correct checksum
     Given using old dav path
+    And user "user0" exists
+    And file "prueba_cksum.txt" with text "Test file for checksums" is created in local storage
+    When user "user0" downloads the file "/local_storage/prueba_cksum.txt"
+    When user "user0" downloads the file "/local_storage/prueba_cksum.txt"
+    Then The header checksum should match "SHA1:a35b7605c8f586d735435535c337adc066c2ccb6"
+
+  Scenario: Uploading a file with checksum should work using new dav path
+    Given using new dav path
+    And user "user0" exists
+    When user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
+    Then The webdav response should have a status code "201"
+
+  Scenario: Uploading a file with checksum should return the checksum in the propfind using new dav path
+    Given using new dav path
+    And user "user0" exists
+    And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
+    When user "user0" request the checksum of "/myChecksumFile.txt" via propfind
+    Then The webdav checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
+
+  Scenario: Uploading a file with checksum should return the checksum in the download header using new dav path
+    Given using new dav path
+    And user "user0" exists
+    And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
+    When user "user0" downloads the file "/myChecksumFile.txt"
+    Then The header checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
+
+  Scenario: Moving a file with checksum should return the checksum in the propfind using new dav path
+    Given using new dav path
+    And user "user0" exists
+    And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
+    When User "user0" moved file "/myChecksumFile.txt" to "/myMovedChecksumFile.txt"
+    And user "user0" request the checksum of "/myMovedChecksumFile.txt" via propfind
+    Then The webdav checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
+
+  Scenario: Moving file with checksum should return the checksum in the download header using new dav path
+    Given using new dav path
+    And user "user0" exists
+    And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
+    When User "user0" moved file "/myChecksumFile.txt" to "/myMovedChecksumFile.txt"
+    And user "user0" downloads the file "/myMovedChecksumFile.txt"
+    Then The header checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
+
+  Scenario: Copying a file with checksum should return the checksum in the propfind using new dav path
+    Given using new dav path
+    And user "user0" exists
+    And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
+    When User "user0" copies file "/myChecksumFile.txt" to "/myChecksumFileCopy.txt"
+    And user "user0" request the checksum of "/myChecksumFileCopy.txt" via propfind
+    Then The webdav checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
+
+  Scenario: Copying file with checksum should return the checksum in the download header using new dav path
+    Given using new dav path
+    And user "user0" exists
+    And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
+    When User "user0" copies file "/myChecksumFile.txt" to "/myChecksumFileCopy.txt"
+    And user "user0" downloads the file "/myChecksumFileCopy.txt"
+    Then The header checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
+
+  Scenario: Uploading a chunked file with checksum should return the checksum in the propfind using new dav path
+    Given using new dav path
+    And user "user0" exists
+    And user "user0" uploads chunk file "1" of "3" with "AAAAA" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
+    And user "user0" uploads chunk file "2" of "3" with "BBBBB" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
+    And user "user0" uploads chunk file "3" of "3" with "CCCCC" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
+    When user "user0" request the checksum of "/myChecksumFile.txt" via propfind
+    Then The webdav checksum should match "SHA1:acfa6b1565f9710d4d497c6035d5c069bd35a8e8"
+
+  Scenario: Uploading a chunked file with checksum should return the checksum in the download header using new dav path
+    Given using new dav path
+    And user "user0" exists
+    And user "user0" uploads chunk file "1" of "3" with "AAAAA" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
+    And user "user0" uploads chunk file "2" of "3" with "BBBBB" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
+    And user "user0" uploads chunk file "3" of "3" with "CCCCC" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
+    When user "user0" downloads the file "/myChecksumFile.txt"
+    Then The header checksum should match "SHA1:acfa6b1565f9710d4d497c6035d5c069bd35a8e8"
+
+  Scenario: Downloading a file from local storage has correct checksum using new dav path
+    Given using new dav path
     And user "user0" exists
     And file "prueba_cksum.txt" with text "Test file for checksums" is created in local storage
     When user "user0" downloads the file "/local_storage/prueba_cksum.txt"

--- a/tests/integration/features/checksums.feature
+++ b/tests/integration/features/checksums.feature
@@ -139,3 +139,36 @@ Feature: checksums
     When user "user0" downloads the file "/local_storage/prueba_cksum.txt"
     When user "user0" downloads the file "/local_storage/prueba_cksum.txt"
     Then The header checksum should match "SHA1:a35b7605c8f586d735435535c337adc066c2ccb6"
+
+  Scenario: Upload chunked file where checksum does not match
+    Given using new dav path
+    And user "user0" exists
+    And user "user0" creates a new chunking upload with id "chunking-42"
+    And user "user0" uploads new chunk file "2" with "BBBBB" to id "chunking-42"
+    And user "user0" uploads new chunk file "3" with "CCCCC" to id "chunking-42" with checksum "SHA1:f005ba11"
+    And user "user0" uploads new chunk file "1" with "AAAAA" to id "chunking-42"
+    And user "user0" moves new chunk file with id "chunking-42" to "/myChunkedFile.txt"
+    Then the HTTP status code should be "400"
+
+  Scenario: Upload a file where checksum does not match
+    Given using old dav path
+    Given user "user0" exists
+    And file "/chksumtst.txt"  does not exist for user "user0"
+    And user "user0" uploads file with checksum "SHA1:f005ba11" and content "Some Text" to "/chksumtst.txt"
+    Then the HTTP status code should be "400"
+
+  Scenario: Upload a file where checksum does match
+    Given using old dav path
+    Given user "user0" exists
+    And file "/chksumtst.txt"  does not exist for user "user0"
+    And user "user0" uploads file with checksum "SHA1:ce5582148c6f0c1282335b87df5ed4be4b781399" and content "Some Text" to "/chksumtst.txt"
+    Then the HTTP status code should be "201"
+
+  Scenario: Uploaded file should have the same checksum when downloaded
+    Given using old dav path
+    Given user "user0" exists
+    And file "/chksumtst.txt"  does not exist for user "user0"
+    And user "user0" uploads file with checksum "SHA1:ce5582148c6f0c1282335b87df5ed4be4b781399" and content "Some Text" to "/chksumtst.txt"
+    When Downloading file "/chksumtst.txt" as "user0"
+    Then The following headers should be set
+            | OC-Checksum | SHA1:ce5582148c6f0c1282335b87df5ed4be4b781399 |

--- a/tests/integration/features/checksums.feature
+++ b/tests/integration/features/checksums.feature
@@ -1,52 +1,60 @@
 Feature: checksums
 
   Scenario: Uploading a file with checksum should work
-    Given user "user0" exists
+    Given using old dav path
+    And user "user0" exists
     When user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
     Then The webdav response should have a status code "201"
 
   Scenario: Uploading a file with checksum should return the checksum in the propfind
-    Given user "user0" exists
+    Given using old dav path
+    And user "user0" exists
     And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
     When user "user0" request the checksum of "/myChecksumFile.txt" via propfind
     Then The webdav checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
 
   Scenario: Uploading a file with checksum should return the checksum in the download header
-    Given user "user0" exists
+    Given using old dav path
+    And user "user0" exists
     And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
     When user "user0" downloads the file "/myChecksumFile.txt"
     Then The header checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
 
   Scenario: Moving a file with checksum should return the checksum in the propfind
-    Given user "user0" exists
+    Given using old dav path
+    And user "user0" exists
     And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
     When User "user0" moved file "/myChecksumFile.txt" to "/myMovedChecksumFile.txt"
     And user "user0" request the checksum of "/myMovedChecksumFile.txt" via propfind
     Then The webdav checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
 
   Scenario: Moving file with checksum should return the checksum in the download header
-    Given user "user0" exists
+    Given using old dav path
+    And user "user0" exists
     And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
     When User "user0" moved file "/myChecksumFile.txt" to "/myMovedChecksumFile.txt"
     And user "user0" downloads the file "/myMovedChecksumFile.txt"
     Then The header checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
 
   Scenario: Copying a file with checksum should return the checksum in the propfind
-    Given user "user0" exists
+    Given using old dav path
+    And user "user0" exists
     And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
     When User "user0" copied file "/myChecksumFile.txt" to "/myChecksumFileCopy.txt"
     And user "user0" request the checksum of "/myChecksumFileCopy.txt" via propfind
     Then The webdav checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
 
   Scenario: Copying file with checksum should return the checksum in the download header
-    Given user "user0" exists
+    Given using old dav path
+    And user "user0" exists
     And user "user0" uploads file "data/textfile.txt" to "/myChecksumFile.txt" with checksum "MD5:d70b40f177b14b470d1756a3c12b963a"
     When User "user0" copied file "/myChecksumFile.txt" to "/myChecksumFileCopy.txt"
     And user "user0" downloads the file "/myChecksumFileCopy.txt"
     Then The header checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
 
   Scenario: Uploading a chunked file with checksum should return the checksum in the propfind
-    Given user "user0" exists
+    Given using old dav path
+    And user "user0" exists
     And user "user0" uploads chunk file "1" of "3" with "AAAAA" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
     And user "user0" uploads chunk file "2" of "3" with "BBBBB" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
     And user "user0" uploads chunk file "3" of "3" with "CCCCC" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
@@ -54,7 +62,8 @@ Feature: checksums
     Then The webdav checksum should match "SHA1:acfa6b1565f9710d4d497c6035d5c069bd35a8e8"
 
   Scenario: Uploading a chunked file with checksum should return the checksum in the download header
-    Given user "user0" exists
+    Given using old dav path
+    And user "user0" exists
     And user "user0" uploads chunk file "1" of "3" with "AAAAA" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
     And user "user0" uploads chunk file "2" of "3" with "BBBBB" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
     And user "user0" uploads chunk file "3" of "3" with "CCCCC" to "/myChecksumFile.txt" with checksum "MD5:45a72715acdd5019c5be30bdbb75233e"
@@ -62,7 +71,8 @@ Feature: checksums
     Then The header checksum should match "SHA1:acfa6b1565f9710d4d497c6035d5c069bd35a8e8"
 
   Scenario: Downloading a file from local storage has correct checksum
-    Given user "user0" exists
+    Given using old dav path
+    And user "user0" exists
     And file "prueba_cksum.txt" with text "Test file for checksums" is created in local storage
     When user "user0" downloads the file "/local_storage/prueba_cksum.txt"
     When user "user0" downloads the file "/local_storage/prueba_cksum.txt"

